contracts/contracts/lib/LibStaking.sol

```diff
@@ -583,15 +583,10 @@ library LibStaking {
                     IGateway(gateway).releaseStake(amount);
                 } else if (change.op == StakingOperation.Deposit)  {
                     s.validatorSet.confirmDeposit(validator, amount);
-
-                    if (s.collateralSource.isNative()) {
-                        IGateway(gateway).addStake{value: amount}(amount);
-                    } else {
-                        s.collateralSource.increaseAllowance(gateway, amount);
-                        IGateway(gateway).addStake(amount);
-                    }
+                    uint256 msgValue = s.collateralSource.makeAvailable(gateway, amount);
+                    IGateway(gateway).addStake{value: amount}(amount);
```

Comment:
I think this is not correct, it should be:
```
IGateway(gateway).addStake{value: msgValue}(amount);
```
Pushed a PR to this.

Reply:
Well spotted! Copy pasta